modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala

```diff
@@ -185,12 +219,27 @@ object K1 {
       case _ => EmptyTuple
     }
 
+  /**
+   * Summon the first given instance `F[U]` from the tuple `T`.
+   * Remaining elements of `T` may or may not have an instance of `F`.
+   */
   inline def summonFirst[F[_[_]], T[_], U[_]]: F[U] =
     K0.summonFirst0[LiftP[F, T]].asInstanceOf[F[U]]
 
   transparent inline def summonFirst0[T]: Any =
     K0.summonFirst0[T]
 
+  /**
+   * Summon the only given instance `F[U]` from the tuple `T`.
+   * Remaining elements of `T` are guaranteed to not have an instance of `F`.
+   */
+  inline def summonOnly[F[_[_]], T[_], U[_]]: F[U] =
+    K0.summonOnly0[LiftP[F, T]].asInstanceOf[F[U]]
```

Comment:
Should these methods be extracted to somewhere other than `K0` as they aren't specific to a given kind?

Reply:
Idk, `summonFirst0` is already in `K0` and we can't move it now 🤷 
Do you have any better idea? In general I agree and even `summonFirst0` is present multiple times 😬

Reply:
@TimWSpence I moved them to an object in the `internals` package 👍 
I think that's much better now. Also deprecated the old helpers.